internal/sync/sync.go

```diff
@@ -261,6 +263,46 @@ const (
 	maxAcceptableDispatch = time.Millisecond * 50
 )
 
+// Returns remote protocol version as a string.
+// Or, an error if one exists.
+func (s Sync) Version() (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	conn, err := s.Client.DialWsep(ctx, s.Env)
+	if err != nil {
+		return "", err
+	}
+	defer conn.Close(websocket.CloseNormalClosure, "")
+
+	execer := wsep.RemoteExecer(conn)
+	process, err := execer.Start(ctx, wsep.Command{
+		Command: "rsync",
+		Args:    []string{"--version"},
+	})
+	if err != nil {
+		return "", err
+	}
+	r := bufio.NewReader(process.Stdout())
```

Comment:
We have the same issue here with the reader never reading anything.

I think the best solution here would be to replace this reader with:
```
buf := &bytes.Buffer{}
io.Copy(buf, process.Stdout())
```

And `buf` can be used down below to read the first line.

Reply:
Will do.

Reply:
Do you want the remote check to `log.Fatal` fail if we can't retrieve it?

Reply:
I think I prefer the behavior you had, where we just return the error and warn the user.

Reply:
The remote check still has that.